messaging/src/main/java/org/axonframework/eventhandling/pooled/WorkPackage.java

```diff
@@ -84,6 +84,7 @@ class WorkPackage {
     private final int batchSize;
     private final long claimExtensionThreshold;
     private final Consumer<UnaryOperator<TrackerStatus>> segmentStatusUpdater;
+    private final Consumer<Integer> segmentEventsBatchProcessedCallback;
```

Comment:
Another rename suggestion. I think just calling this object the `batchProcessedCallback` is sufficient. In the context of a `WorkPackage` you'd always be dealing with (1) a single segment and (2) with events. Hence, we can omit those from the variable name.

Reply:
I tried to be (at least to some point) consistent with the name above **segment**StatusUpdater. 
But `batchProcessedCallback` sounds good to me.